// components/HamburgerEmphatic/StyledLinesEmphatic.js
import { StyledLines } from '../styledLines/StyledLines'
import { getBarColor } from '../functions/'

const before = `
  left: 0;
  transition: transform 0.125s cubic-bezier(0.6, 0.04, 0.98, 0.335),
  top 0.05s 0.125s linear, left 0.125s 0.175s ease-in;
`

const isActiveBefore = `
  left: -80px;
  top: -80px;
  transform: translate3d(80px, 80px, 0) rotate(45deg);
  transition: left 0.125s ease-out, top 0.05s 0.125s linear, transform 0.125s 0.175s cubic-bezier(0.075, 0.82, 0.165, 1);
`

const after = `
  top: 10px;
  right: 0;
  transition: transform 0.125s cubic-bezier(0.6, 0.04, 0.98, 0.335),
  top 0.05s 0.125s linear, right 0.125s 0.175s ease-in;
`

const isActiveAfter = `
  right: -80px;
  top: -80px;
  transform: translate3d(-80px, 80px, 0) rotate(-45deg);
  transition: right 0.125s ease-out, top 0.05s 0.125s linear, transform 0.125s 0.175s cubic-bezier(0.075, 0.82, 0.165, 1);
`

export const StyledLinesEmphatic = StyledLines.extend`
  &::before {
    ${before};
    ${({ isActive }) => isActive && isActiveBefore};
    ${getBarColor};
  }
  &::after {
    ${after};
    ${({ isActive }) => isActive && isActiveAfter};
    ${getBarColor};
  }
  transition: background-color 0.125s 0.175s ease-in;
  ${getBarColor};
  ${({ isActive }) =>
    isActive &&
    'transition-delay: 0s; transition-timing-function: ease-out; background-color: transparent;'};
`

StyledLinesEmphatic.displayName = 'StyledLinesEmphatic'
